lib/rubocop/cop/rails/date_and_time_column_name.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop enforces date/time columns naming conventions.
+      # Specifically, `date` column should end with `_on` suffix,
+      # `datetime` and `timestamp` - with `_at` suffix,
+      # `time` - with `_time` suffix.
+      #
+      # @example
+      #   # bad
+      #   add_column :orders, :created_on, :datetime
+      #   t.column :visited_at, :date
+      #   t.time :start_at
+      #
+      #   # good
+      #   add_column :orders, :created_at, :datetime
+      #   t.column :visited_on, :date
+      #   t.time :start_time
+      #
+      class DateAndTimeColumnName < Base
+        MSG = 'Name `%<type>s` columns with `%<suffix>s` suffixes.'
+        TYPE_TO_SUFFIX = {
+          datetime: '_at',
+          timestamp: '_at',
+          date: '_on',
+          time: '_time'
+        }.freeze
+
+        def on_send(node)
+          return unless in_migration?(node)
+
+          column_name, type = column_name_and_type(node)
+          return unless column_name
+
+          suffix = TYPE_TO_SUFFIX[type]
+          return if !suffix || column_name.value.to_s.end_with?(suffix)
+
+          message = format(MSG, type: type, suffix: suffix)
+          add_offense(column_name, message: message)
+        end
+
+        private
+
+        def_node_matcher :migration_base?, <<~PATTERN
+          {
+            (send (const (const nil? :ActiveRecord) :Migration) :[] _)
+            (const (const nil? :ActiveRecord) :Migration)
```

Comment:
Wondering if some projects use a common `ApplicationMigration` 🤔 
Is this check really necessary if we already know that we're inspecting `db/migrations/*.rb`?

Reply:
With this change
```
 Include:
    - db/**/*.rb
```
we should at least ignore schema.rb files. It is also possible (but very unlikely) to have some other files there, except migration files. Better to double check, I think.